internal/provider/exception_resource.go

```diff
@@ -0,0 +1,268 @@
+	// Extract ScopeMrn
+	scopeMrn = data.ScopeMrn.ValueString()
+	if scopeMrn == "" {
+		scopeMrn = r.client.space.MRN()
+	}
+
+	// Extract Checks and Vulnerabilities
+	checks = []string{}
+	data.CheckMrns.ElementsAs(ctx, &checks, false)
+
+	vulnerabilities = []string{}
+	data.VulnerabilityMrns.ElementsAs(ctx, &vulnerabilities, false)
+
+	// Format ValidUntil to RFC3339 if provided
+	validUntil := data.ValidUntil.ValueString()
+	if validUntil != "" {
+		year, month, day, parseErr := parseDate(validUntil)
+		if parseErr != nil {
+			return "", nil, nil, "", parseErr
+		}
+		now := time.Now().UTC() // Use UTC directly
+		validUntilStr = time.Date(
+			year,
+			month,
+			day,
+			now.Hour(),
+			now.Minute(),
+			now.Second(),
+			now.Nanosecond(),
+			time.UTC,
+		).Format(time.RFC3339Nano) // Use RFC3339Nano to include nanoseconds
+	}
+
+	return scopeMrn, checks, vulnerabilities, validUntilStr, nil
+}
+
+func (r *exceptionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `Set custom exceptions fot a Scope.`,
+		Attributes: map[string]schema.Attribute{
+			"scope_mrn": schema.StringAttribute{
+				MarkdownDescription: "The MRN of the scope (either asset mrn or space mrn).",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"valid_until": schema.StringAttribute{
+				MarkdownDescription: "The timestamp until the exception is valid.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`[1-9][0-9][0-9]{2}-([0][1-9]|[1][0-2])-([1-2][0-9]|[0][1-9]|[3][0-1])`), "Date must be in the format 'YYYY-MM-DD'"),
+				},
+			},
+			"justification": schema.StringAttribute{
+				MarkdownDescription: "Description why the exception is required.",
+				Optional:            true,
+			},
+			"action": schema.StringAttribute{
+				MarkdownDescription: "The action to perform. Default is `SNOOZE`. Other options are `ENABLE`, `DISABLE`, `OUT_OF_SCOPE`.",
+				Optional:            true,
+				Computed:            true,
+				Default:             stringdefault.StaticString("SNOOZE"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("SNOOZE", "ENABLE", "DISABLE", "OUT_OF_SCOPE"),
+				},
+			},
+			"check_mrns": schema.ListAttribute{
+				MarkdownDescription: "List of check MRNs to set exceptions for.",
+				ElementType:         types.StringType,
+				Optional:            true,
+			},
+			"vulnerability_mrns": schema.ListAttribute{
+				MarkdownDescription: "List of vulnerability MRNs to set exceptions for.",
+				ElementType:         types.StringType,
+				Optional:            true,
+			},
```

Comment:
how come everything is optional? what happens if users add an empty resourse?

Reply:
I do not see how we are supposed to make them Required. I therefore added some new validations to help the user decide when to set which attribute